### ISSUE TITLE
Incoming new device connections are ignored

### ISSUE THREAD

**Issue by nekr0z** [syncthing/syncthing-lite#138]
Syncthing Lite just ignores any remote device that try to share folders. This defeats the whole introducer device idea in Syncthing.

**Comment by l-jonas** [comment]
So the issue is "feature request: add introducer support"?

**Comment by nekr0z** [comment]
Not really. Proper introducer support should perhaps include allowing to mark a device as introducer in Syncthing Lite. IMO that would be the next step, and a separate issue.

This issue is strictly about pairing process, i.e.:
1. Selects "show my ID" on device A and writes down the device ID.
2. Select "add a new device" on device B and put there the ID from step 1.
3. Device B tries to connect to device A.
4. Device A shown notification "a new device (B) is trying to connect, accept?" Accept.
5. Now the devices are connected, folders can be shared, everything is OK.

Currently it works as expected if device B is a Syncthing Lite instance. But if device A is a Syncthing Lite instance, step 4 never happens.

**Comment by l-jonas** [comment]
Syncthing Lite does not support incoming connections at all (and I'm not sure of any case where incoming connections are useful). But I could think of making it easier to add devices by showing ids from the local network (from the local discovery).

**Comment by nekr0z** [comment]
Here is a usecase for you to consider:

- user has a laptop, a workstation at home, a workstation at her office, a couple of servers somewhere it the cloud;
- all the user's computers run Syncthing and have user's laptop as introducer device;
- user has a folder shared among all devices (presumably some documents she's working on, or a collection of PDFs on her scientific field, or whatever);
- user wants to access this folder via Syncthing Lite, so she adds her laptop as a device and shares the folder from the laptop

Now, all her computers already know about her Syncthing Lite instance and try to connect to it. Sadly, Syncthing Lite does not support it and ignores them, because only the laptop was explicitly added. Ideally our user would want Syncthing Lite to be able to connect to any of her devices: laptop is not always on, but servers are, and workstations are accessible locally at home and in the office for faster file transfers.

Currently, our user needs to add all her computers manually. And even if you do show IDs from local discovery, she would need to go and fetch IDs of her servers from somewhere. A lot of hassle, bad UX.

Imagine if ST Lite supported incoming connections. All the user's computers would almost immediately appear as incoming connection offers, and she would only need to touch "accept" button next to each of them. Much less hassle, much better UX.

**Comment by l-jonas** [comment]
Well, in this case, the laptop would introduce the other devices to syncthing lite, so it would be better to just add the introduction support to Syncthing Lite.

> workstations are accessible locally at home and in the office for faster file transfers

I'm not sure if the current implementation does this as good as possible.

**Comment by nekr0z** [comment]
> it would be better to just add the introduction support

True, that would be even more awesome.

> the laptop would introduce the other devices to syncthing lite

In this issue I rather meant that it already introduces ST Lite to other devices, but they can't make use of this introduction.
